prog/dftb+/lib_math/lapackroutines.F90

```diff
@@ -1155,6 +1157,111 @@ subroutine larnv_dblecplx(iDist,iSeed,x)
   end subroutine larnv_dblecplx
 
 
+  !> real svd decomposition of matrix A into left and right vectors and singular values
+  subroutine sgesvd_real(A,u,sigma,vt)
+
+    !> matrix to decompose, warning the matrix is over-written by the routine
+    real(rsp), intent(inout) :: A(:,:)
+
+    !> first min(m,n) columns of u hold the left singular vector on return
+    real(rsp), intent(out) :: u(:,:)
+
+    !> holds the singular values on return
+    real(rsp), intent(out) :: sigma(:)
+
+    !> first min(m,n) columns of vt hold the right singular vector on return - warning this matrix
+    !> is returned transpose(conjugated()) i.e. A = u.s.vt and all non-returned singular vectors are
+    !> zero!
+    real(rsp), intent(out) :: vt(:,:)
+
+    integer :: n, m, mn, lda, lwork, ldu, ldvt, info
+    real(rsp), allocatable :: work(:)
+
+    m = size(A,dim=1)
+    n = size(A,dim=2)
+    mn = min(m,n)
+    lda = size(A,dim=1)
+    ldu = size(U,dim=1)
+    ldvt = size(Vt,dim=1)
+    @:ASSERT(all(shape(u) >= (/m,mn/)))
+    @:ASSERT(all(shape(vt) >= (/mn,n/)))
+    @:ASSERT(size(sigma) == mn)
```

Comment:
Why `>=` instead of `==`? According to LAPACK no elements beyond `[m, mn]` or `[mn, m]` are set within the call, so it would be logical to ask for exactly that array size.

Reply:
Not if you have jobu/jobvt = 'A' instead of 'S'.